### ISSUE TITLE
Web Deploy task exclude files / folders

### ISSUE THREAD

**Issue by swcarter007** [Microsoft/vso-agent-tasks#1215]
Right now, the web deploy task removes folders or files that are not part of the build.  This is causing issues with our logging files that we want to retain as well as images that are placed on the server from a CMS system.

**Comment by bryanmacfarlane** [comment]
We need more details about the build definitions, tasks and path examples.

**Comment by swcarter007** [comment]
Sure.  I am using the IIS Web App Deployment tasks to take a pre-built .zip file that was generated from a MSBuild task.  I want to take that zip file and deploy it to a specific file on an on-prem server, and it is deleting out the subfolders within the location I am deploying to.  So for my example, I am trying to deploy to D:/Websites/uat_coll where I am running my web application, and it is deleting out the folders at D:/Websites/uat_coll/images even though that folder doesn't exist in the bundled zip file that was created by the msbuild task during the build phase of the steps.

**Comment by mvvsubbu** [comment]
As the intent of the task is to deploy to web deploy package to the specified location. We use msdeploy -verb:sync option to do that and this command extracts the zip and makes sure that target folder is in sync with source zip, it is expected behavior.  

Can you help me understand why the images folder is static in that folder, instead this should have been checked into to source control along with web project, so that zip will contain this.

**Comment by mvvsubbu** [comment]
https://github.com/Microsoft/vso-agent-tasks/issues/1215

**Comment by mvvsubbu** [comment]
We will address this feedback in next version, for now you can pass -enableRule:DoNotDeleteRule in override params section of the task

**Comment by mvvsubbu** [comment]
Can I go ahead and close the issue?

**Comment by swcarter007** [comment]
Sure, thank you. 

----- Original Message -----

**Comment by mvvsubbu** [comment]
We are creating an extension for this task soon, We will add this feature in that version of task. Thanks for your feedback, we will track it, you can refer above commit to unblock your self
